pkg/apis/core/pods/helpers.go

```diff
@@ -0,0 +1,51 @@
+/*
+Copyright 2017 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package pods
+
+import (
+	"fmt"
+
+	"k8s.io/kubernetes/pkg/fieldpath"
+)
+
+func ConvertDownwardAPIFieldLabel(version, label, value string) (string, string, error) {
+	if version != "v1" {
+		return "", "", fmt.Errorf("unsupported pod version: %s", version)
+	}
+
+	path, _ := fieldpath.SplitMaybeSubscriptedPath(label)
```

Comment:
same here, separate path for subscripted and non-subscripted would make more sense

```
if path, subscript, ok := fieldpath.SplitSubscriptedPath(label); ok {
  switch path {
    case "metadata.annotations", "metadata.labels":
      return label
    default:
      return error
  }
}

switch label {
  ... existing fixed mapping ...
}
```

Reply:
Done. Please take another look.